app/models/concerns/participant_declaration_event_recorder.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# Currently a placeholder to ensure that transitioning to this doesn't break since the scopes will be different
+# 1. Duplicate ect_profiles can be recorded, so we need to filter for uniqueness as a count
+# 2. Order is reversed, to pick up the latest change to a user
+# 3. Holding "retention_1" rather than "Retention 1" in the db, for example.
+
+module Concerns
+  module ParticipantDeclarationEventRecorder
+    extend ActiveSupport::Concern
+    included do
+      scope :unique_early_career_teacher_profile_id, -> { select(:early_career_teacher_profile_id).distinct }
+      scope :active, -> { where(declaration_type: "Start").order(declaration_date: "desc").unique_early_career_teacher_profile_id }
+      scope :declared_as_between, ->(start_date, end_date) { where(declaration_date: start_date..end_date) }
+      scope :submitted_between, ->(start_date, end_date) { where(created_at: start_date..end_date) }
+    end
+  end
+end
```

Comment:
What do we get out of making it a concern? I can't think of it being reused, and `ParticipantDeclaration` doesn't really strike me as a long file in need of splitting,

Reply:
Reason is that if we swap between the two different mechanisms for either event or cached state holding the "active" in two different concerns means that there is less to change.

Reply:
"If we swap"? Are we not sure? Do we plan to have both mechanisms in our codebase for extended period of time?

Reply:
Yes. We'll probably end up using both. There are use cases where one is more appropriate as cached information and another where we need all information as it is.